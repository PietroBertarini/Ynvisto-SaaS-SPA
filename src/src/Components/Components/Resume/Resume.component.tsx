import React from 'react';
import Box from '@material-ui/core/Box';

import { useStyles } from './Resume.styles';
import ResumeInfo, { ResumoInfoProps } from '../ResumeInfo/ResumeInfo.component';

export interface ResumeProps {
    /**
     * Array with all resume info to show at page. Max 3
     */
    allResumeInfo: ResumoInfoProps[];

}
/**
 * Simple component that have a value in R$ format with status color and a subtitle.
 */
function Resume(props :ResumeProps) {
  const {
    allResumeInfo,
  } = props;
  const classes = useStyles();
  if (allResumeInfo.length === 1) {
    return (
      <Box display="flex" flexDirection="row" justifyContent="space-between" className={classes.resume}>
        <Box />
        <ResumeInfo
          value={allResumeInfo[0].value}
          status={allResumeInfo[0].status}
          subtitle={allResumeInfo[0].subtitle}
        />
        <Box />
      </Box>
    );
  }
  if (allResumeInfo.length === 2) {
    return (
      <Box display="flex" flexDirection="row" justifyContent="space-between" className={classes.resume}>
        <ResumeInfo
          value={allResumeInfo[0].value}
          status={allResumeInfo[0].status}
          subtitle={allResumeInfo[0].subtitle}
        />
        <Box />
        <ResumeInfo
          value={allResumeInfo[1].value}
          status={allResumeInfo[1].status}
          subtitle={allResumeInfo[1].subtitle}
        />
      </Box>
    );
  }
  if (allResumeInfo.length >= 3) {
    return (
      <Box display="flex" flexDirection="row" justifyContent="space-between" className={classes.resume}>
        <ResumeInfo
          value={allResumeInfo[0].value}
          status={allResumeInfo[0].status}
          subtitle={allResumeInfo[0].subtitle}
        />
        <ResumeInfo
          value={allResumeInfo[1].value}
          status={allResumeInfo[1].status}
          subtitle={allResumeInfo[1].subtitle}
        />
        <ResumeInfo
          value={allResumeInfo[2].value}
          status={allResumeInfo[2].status}
          subtitle={allResumeInfo[2].subtitle}
        />
      </Box>
    );
  }
  return (
    <div />
  );
}
export default Resume;
