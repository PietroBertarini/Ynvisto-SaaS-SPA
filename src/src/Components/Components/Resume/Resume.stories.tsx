import React from 'react';
// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
import { Meta, Story } from '@storybook/react/types-6-0';
import Resume, { ResumeProps } from './Resume.component';
import {
  mockOneResumeInfo,
  mockThreeResumeInfo,
  mockTwoResumeInfo,
} from '../../../Mocks/ResumeInfo.mock';

export default {
  title: 'Components/Resume',
  component: Resume,
} as Meta;

const Template: Story<ResumeProps> = (args) => <Resume {...args} />;

export const Three = Template.bind({});
Three.args = {
  allResumeInfo: mockThreeResumeInfo,
};
export const Two = Template.bind({});
Two.args = {
  allResumeInfo: mockTwoResumeInfo,
};
export const One = Template.bind({});
One.args = {
  allResumeInfo: mockOneResumeInfo,
};
