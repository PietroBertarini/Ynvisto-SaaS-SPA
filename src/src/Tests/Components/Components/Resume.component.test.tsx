import React from 'react';
import { shallow } from 'enzyme';
import Resume from '../../../Components/Components/Resume/Resume.component';
import {
  mockOneResumeInfo, mockThreeResumeInfo, mockTwoResumeInfo,
} from '../../../Mocks/ResumeInfo.mock';

test('Renders Resume components and match to previously snapshot', () => {
  const oneStatusComponent = shallow(
    <Resume allResumeInfo={mockOneResumeInfo} />,
  );
  const twoStatusComponent = shallow(
    <Resume allResumeInfo={mockTwoResumeInfo} />,
  );
  const threeStatusComponent = shallow(
    <Resume allResumeInfo={mockThreeResumeInfo} />,
  );
  const fiveStatusComponent = shallow(
    <Resume allResumeInfo={mockThreeResumeInfo.concat(mockTwoResumeInfo)} />,
  );

  expect(oneStatusComponent).toMatchSnapshot();
  expect(twoStatusComponent).toMatchSnapshot();
  expect(threeStatusComponent).toMatchSnapshot();
  expect(fiveStatusComponent).toMatchSnapshot();
});
