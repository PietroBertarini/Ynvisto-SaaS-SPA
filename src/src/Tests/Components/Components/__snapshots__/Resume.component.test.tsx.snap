// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Resume components and match to previously snapshot 1`] = `
<Styled(MuiBox)
  className="makeStyles-resume-1"
  display="flex"
  flexDirection="row"
  justifyContent="space-between"
>
  <Styled(MuiBox) />
  <ResumeInfo
    status="WARNING"
    subtitle="Volume isento disponível em Junho"
    value={30304.94}
  />
  <Styled(MuiBox) />
</Styled(MuiBox)>
`;

exports[`Renders Resume components and match to previously snapshot 2`] = `
<Styled(MuiBox)
  className="makeStyles-resume-1"
  display="flex"
  flexDirection="row"
  justifyContent="space-between"
>
  <ResumeInfo
    status="BAD"
    subtitle="Volume isento disponível em Maio"
    value={30304.94}
  />
  <Styled(MuiBox) />
  <ResumeInfo
    status="WARNING"
    subtitle="Volume isento disponível em Maio"
    value={30304.94}
  />
</Styled(MuiBox)>
`;

exports[`Renders Resume components and match to previously snapshot 3`] = `
<Styled(MuiBox)
  className="makeStyles-resume-1"
  display="flex"
  flexDirection="row"
  justifyContent="space-between"
>
  <ResumeInfo
    status="GOOD"
    subtitle="Volume isento disponível em Maio"
    value={30304.94}
  />
  <ResumeInfo
    status="BAD"
    subtitle="Volume isento disponível em Maio"
    value={30304.94}
  />
  <ResumeInfo
    status="WARNING"
    subtitle="Volume isento disponível em Maio"
    value={30304.94}
  />
</Styled(MuiBox)>
`;

exports[`Renders Resume components and match to previously snapshot 4`] = `
<Styled(MuiBox)
  className="makeStyles-resume-1"
  display="flex"
  flexDirection="row"
  justifyContent="space-between"
>
  <ResumeInfo
    status="GOOD"
    subtitle="Volume isento disponível em Maio"
    value={30304.94}
  />
  <ResumeInfo
    status="BAD"
    subtitle="Volume isento disponível em Maio"
    value={30304.94}
  />
  <ResumeInfo
    status="WARNING"
    subtitle="Volume isento disponível em Maio"
    value={30304.94}
  />
</Styled(MuiBox)>
`;
